docs/manual/constant-value-checker.tex

```diff
@@ -334,30 +334,30 @@
 that possibility, or verify the lack of overflow using some other tool or
 manual analysis.
 
-\sectionAndLabel{Strings can be null in concatenations}{null-strings-concats}
 
-By default, the Constant Value Checker is unsound with respect to string
-concatenation.  The \code{-AnullStringsConcatenation} command-line option
-eliminates this unsoundness, though it may lead to false positive warnings.
+\sectionAndLabel{Strings can be null in concatenations}{non-null-strings-concats}
+
+By default, the Constant Value Checker is sound with respect to string
+concatenation and nullness.  It assumes that, in a string concatenation,
+every non-primitive argument might be null, except for String literals
+and compile-time constants. It ignores Nullness Checker annotations.
```

Comment:
1. String literals are a type of compile-time constant, so I don't think they need to be called out explicitly.
2. I don't see any test cases for compile-time constants that are not string literals.  Could you add some?  It might be enough to duplicate the `nullConcatenation` test (in both places it occurs) and change the local variables to `final` in the copy.